vassal-doc/src/main/readme-referencemanual/ReferenceManual/Search.adoc

```diff
@@ -18,32 +24,33 @@ The default search is to search for everything, but the Advanced Searh options c
 [width="100%",cols="50%a,^50%a",]
 |===
 a|
+
+===== Search Type
+Search may be normal, start of word or https://en.wikipedia.org/wiki/Regular_expression[Regular Expression]. "Start of word" uses a pre-formatted regular expression to find matches on words or phrases starting with the search string.
+
 ===== Exact Case
 Force an Exact case search (i.e. abc is different to ABC).
 
-===== Regular Expression
-The specified search string is a https://en.wikipedia.org/wiki/Regular_expression[Regular Expression]. In the simplest example, a search for ABC will find matches that equate wholly to "ABC", and lower or mixed case variants (unless _Exact Case_ is checked as well).
-
-===== Advanced Search
-Show the advanced search options. All search options are on in a simple search. The advanced options allow you to turn off options to narrow your search.
+===== Advanced Filters
+Show the advanced filter options. All search filters are on in a simple search. The advanced options allow you to de-select various categories to narrow your search.
```

Comment:
I'm not sure "All search filters are on in a simple search" is actually true? I think simple search maybe only searches for names/types of components and doesn't e.g. search into traits and stuff? I can't remember exactly but that's my impression.

Reply:
> I'm not sure "All search filters are on in a simple search" is actually true?
This is what the doc always said but the help button didn't work until the last vassal release, so perhaps that is moot. 

> I think simple search maybe only searches for names/types of components...
Yes, that is right. However, isn't it natural for a user to assume that search finds all it can by default? This latest PR does that in its current state.  If you think that names & types are the natural filters, then how about I make them the default for "advanced filters" at a restart of the edit session ?

Reply:
We definitely don't want the advanced filters defaulting back to fewer options at an edit session restart -- advanced users are likely to usually want them to stay checked, doubly so since that's been their default. (There's an argument for making the advanced filters remember their settings from session to session within the same module -- but if we don't already have that it would obviously require some work)

Changing the way a simple search operates would have the following challenges:
* Novice users potentially confused by searches into "inner bits" rather than the basic names (I don't have any data on this because we don't collect any; it's just my intuition to give beginners a simpler experience and let them search for things they actually see directly in the edit "tree")
* The code is presently written to ignore the checkboxes if we're not in advanced mode (because obviously we don't want to use checkbox settings that the user can't see and may not even know about), so that code would have to be changed to change the behavior of simple search. 

My preference in this case is definitely to make the documentation match the present behavior, rather than vice versa.

Reply:
> There's an argument for making the advanced filters remember their settings from session to session within the same module -- but if we don't already have that it would obviously require some work

That's what's there; yes, makes sense. The one improvement might be if it could be a global Vassal option.

> * Novice users potentially confused by searches into "inner bits" rather than the basic names (I don't have any data on this because we don't collect any; it's just my intuition to give beginners a simpler experience and let them search for things they actually see directly in the edit "tree")

Novice users will still be interested in "inner bits" like properties and message formats though, surely? There's no telling whether a novice user will want to search on a Name/Class Name or on something they vaguely remember putting in an expression somewhere. As one such user, I can say I've ended up using Advanced as my default option without knowing why until just recently.  The default search position of most apps I have come across is to search for everything and offer the user the ability to filter.

> * The code is presently written to ignore the checkboxes if we're not in advanced mode (because obviously we don't want to use checkbox settings that the user can't see and may not even know about), so that code would have to be changed to change the behavior of simple search.

I already prep'd that change. It meant finding all tests on the "advanced" settings and allowing them if the "Advanced" setting is off.

I would like to get user feedback on the potential change and see if there is a strong preference on any aspect like this one.

Reply:
On second thoughts, let's go with your original design, to which I will add an option for full search.

Reply:
Okay let us know when ready for next review

Reply:
Ready for review. See also PR#12835.